graphql/server.go

```diff
@@ -269,19 +272,26 @@ func (c *conn) handleSubscribe(in *inEnvelope) error {
 			return nil, err
 		}
 
-		d := diff.Diff(previous, current)
+		d := diff.Diff(computationInput.Previous, current)
 		previous = current
-		initial = false
 
-		if initial || d != nil {
+		if d != nil {
 			c.writeOrClose(outEnvelope{
 				ID:       id,
 				Type:     "update",
 				Message:  d,
 				Metadata: output.Metadata,
 			})
+		} else if initial {
+			c.writeOrClose(outEnvelope{
+				ID:       id,
+				Type:     "update",
+				Message:  struct{}{},
```

Comment:
can we change diff to return struct{}{} if there's no diff, and have a unit test that shows merge(any, struct{}{}) returns any?

Reply:
talked offline, we probably don't want to change diff behavior in case there's any external dependency, especially given there's a test case showing the desired behavior of diff(x, x) is nil.

Reply:
on that note, what if we do
```
if d == nil && initial {
  d = struct{}{}
}
```